src/colmap/feature/pairing.cc

```diff
@@ -613,16 +645,14 @@ std::vector<std::pair<image_t, image_t>> SpatialPairGenerator::Next() {
 
   LOG(INFO) << StringPrintf(
       "Matching image [%d/%d]", current_idx_ + 1, position_idxs_.size());
-  const float max_distance =
-      static_cast<float>(options_.max_distance * options_.max_distance);
   for (int j = 0; j < knn_; ++j) {
     // Check if query equals result.
     if (index_matrix_(current_idx_, j) == static_cast<int>(current_idx_)) {
       continue;
     }
 
     // Since the nearest neighbors are sorted by distance, we can break.
-    if (distance_matrix_(current_idx_, j) > max_distance) {
+    if (distance_matrix_(current_idx_, j) > max_distance_squared_) {
```

Comment:
Wouldn't it be easier to change this single line to:
```suggestion
    if (distance_matrix_(current_idx_, j) > max_distance_squared_ && j + 1 >= min_num_neighbors) {
```

Reply:
True, much cleaner. I'll update the PR accordingly.